ArdentHQ/arkscan
# feat: meilisearch

## Summary

Closes https://app.clickup.com/t/861mru5ny

- Added a task for add the official documentation  https://app.clickup.com/t/865cfgn5f
- consider that if `SCOUT_DRIVER=` is set to sometihng different to `meilisearch` is going to use database search, it should work but is quite slow

Note: see deployment instructions below before deployment

### TESTING

1. Run a meilisearch server (you need to use `1.1.1`)
2. Configure your .env settings:

```
SCOUT_DRIVER=meilisearch
MEILISEARCH_HOST=http://127.0.0.1:7700
MEILISEARCH_KEY=
``` 

3. (not sure if neccessary but run) `php artisan scout:sync-index-settings`
4. run (you can run on different tabs at the same time to run in parallel)
  - `php artisan scout:import "App\Models\Wallet"`
  - `php artisan scout:import "App\Models\Transaction"`
  - `php artisan scout:import "App\Models\Block"`
5. it will take a while to index all models but you can copy some of the indexed values from the terminal output to test the results:
![image](https://github.com/ArdentHQ/arkscan/assets/17262776/a6b51571-da15-4405-afc8-d2a220586605)
6. Other considerations:
  - the `scout:import` command call other commands for pause/resume indexing while running since there is a job that will try to find new models every minute, (see source code)

# Deployment

1. First step is to set .env settings 
```
SCOUT_DRIVER=meilisearch
MEILISEARCH_HOST=http://127.0.0.1:7700
MEILISEARCH_KEY=
``` 

2. assuming meilisearch server is running, run `php artisan scout:sync-index-settings`

3. (Indexing: read all points before running the commands)
3.1. run the import commands (you can run on different tabs at the same time to run in parallel)
  - `php artisan scout:import "App\Models\Wallet"`
  - `php artisan scout:import "App\Models\Transaction"`
  - `php artisan scout:import "App\Models\Block"`

3.1. **NOTE:** its going to require some time to add all models, specially with blocks and transactions, maybe a couple of hours or more, while indexing the site should be working as usual but some results may not appear

3.2. Its important to wait until job finishes since the job pauses the scheduled indexing and resumes indexing once finishes, for this you may need to ensure your terminal session is not closed

3.3  We can index everything before releasing in any local machine or other server, you just need to set the env settings to use the production data for meilisearch and database, then run the `php scout:import...` commands. If so you can skip that step once site is deployed. (tip: If you index the database in a local mac machine you can [use owly](https://apps.apple.com/us/app/owly-prevent-display-sleep/id882812218?mt=12) to prevent the machine from sleeping

```
# Production mainet server
ARKSCAN_DB_HOST={production ip}
ARKSCAN_DB_PORT=5432
ARKSCAN_DB_DATABASE={production database}
ARKSCAN_DB_USERNAME={production username}
ARKSCAN_DB_PASSWORD={production password}

SCOUT_DRIVER=meilisearch
# Meilisearch production server
MEILISEARCH_HOST=http://localhost:7777
# Meilisearch production password
MEILISEARCH_KEY=masterkey
```

## Checklist
